lib/posix/pthread_mutex.c

```diff
@@ -22,3 +34,97 @@ int pthread_mutex_trylock(pthread_mutex_t *m)
 
 	return ret;
 }
+
+/**
+ * @brief POSIX threading compatibility API
+ *
+ * See IEEE 1003.1
+ */
+int pthread_mutex_init(pthread_mutex_t *m,
+				     const pthread_mutexattr_t *att)
+{
+	struct k_mem_block block;
+
+	ARG_UNUSED(att);
+
+	if (k_mem_pool_alloc(&posix_mutex_sem, &block,
+			sizeof(struct k_sem), 100) == 0) {
+		memset(block.data, 0, sizeof(struct k_sem));
+	}
+	m->sem = block.data;
+	k_sem_init(m->sem, 1, 1);
+
+	return 0;
+}
+
+/**
+ * @brief POSIX threading compatibility API
+ *
+ * See IEEE 1003.1
+ */
+int pthread_mutex_lock(pthread_mutex_t *m)
+{
+	return k_sem_take(m->sem, K_FOREVER);
+}
+
+/**
+ * @brief POSIX threading compatibility API
+ *
+ * See IEEE 1003.1
+ */
+int pthread_mutex_unlock(pthread_mutex_t *m)
+{
+	k_sem_give(m->sem);
+	return 0;
+}
+
+/**
+ * @brief POSIX threading compatibility API
+ *
+ * See IEEE 1003.1
+ */
+int pthread_mutex_destroy(pthread_mutex_t *m)
+{
+	ARG_UNUSED(m);
```

Comment:
Shouldn't the memory allocated by pthread_mutex_init() be returned to the pool?

Reply:
Done